src/ERPNext/Client.hs

```diff
@@ -18,20 +19,52 @@ module ERPNext.Client
   ) where
 
 import Network.HTTP.Client
-import Data.Text
+import Network.HTTP.Types.Header (hAuthorization, Header)
+import Data.Text hiding (map)
+import Data.Text.Encoding (encodeUtf8)
 import Data.Aeson
 import Data.Proxy qualified
+import ERPNext.Client.Filters (Filters, makeFiltersText)
+import Prelude
 
 -- | Type class for types which represent an ERPNext DocType.
 -- Each DocType has a unique name.
 class IsDocType a where
   docTypeName :: Text
 
-getDocTypeList :: forall a. (IsDocType a, FromJSON a) => Config -> Filters -> ApiResponse [a]
-getDocTypeList = error (unpack $ docTypeName @a)
+getDocTypeList :: forall a. (IsDocType a, FromJSON a) => Config  -> [QueryStringParam]-> IO (ApiResponse [a])
+getDocTypeList config qsParams = do
+  initialRequest <- parseRequest $ unpack $ baseUrl config <> "/resource/" <> docTypeName @a <> "?" <> renderQueryStringParams qsParams
+  let request = initialRequest
+                  { method = "GET"
+                  , requestHeaders = [mkAuthHeader config]
+                  }
+  manager <- Network.HTTP.Client.newManager $ tlsSettings config
+  response <- Network.HTTP.Client.httpLbs request manager
+  let value = decode (responseBody response) :: Maybe Value
+  let result = decode (responseBody response) :: Maybe (DataWrapper [a])
+
+  return $ case (result, value) of
+      (Just res, Just val) -> Ok (getData res) val
+      (Nothing, Just val) -> Err "error" (Just val)
+      (Nothing, Nothing) -> Err "error" Nothing
 
 getDocType :: forall a. (IsDocType a, FromJSON a) => Config -> Text -> IO (ApiResponse a)
-getDocType _ _ = error "implement"
+getDocType config id = do
+    initialRequest <- parseRequest $ unpack $ baseUrl config <> "/resource/" <> docTypeName @a <> "/" <> id
+    let request = initialRequest
+                    { method = "GET"
+                    , requestHeaders = [mkAuthHeader config]
+                    }
+    manager <- Network.HTTP.Client.newManager $ tlsSettings config
+    response <- Network.HTTP.Client.httpLbs request manager
+    let value = decode (responseBody response) :: Maybe Value
+    let result = decode (responseBody response) :: Maybe (DataWrapper a)
```

Comment:
I think we can improve this by not decoding the raw response again but by decoding the `Value`.